server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -1158,9 +1164,24 @@ private void tryToUpdateInsertLastCache(InsertRowPlan plan, Long latestFlushedTi
     }
     IMeasurementMNode[] mNodes = plan.getMeasurementMNodes();
     int columnIndex = 0;
-    for (IMeasurementMNode mNode : mNodes) {
-      // Don't update cached last value for vector type
-      if (!plan.isAligned()) {
+    if (plan.isAligned()) {
+      for (IMeasurementMNode mNode : mNodes) {
+        if (plan.getValues()[columnIndex] == null) {
```

Comment:
How about use the unified for loop way. 
Like `tryToUpdateBatchInsertLastCache(InsertTabletPlan plan, Long latestFlushedTime)`, which use `for (int i = 0; i < mNodes.length; i++) ` ?

Reply:
Fixed